### DISCUSSION TITLE
async validation bug?

### DISCUSSION THREAD

**Discussion by mikemamaril** [vueform/vueform#133] [Questions] [answered]
When setting a unique validator rule as an object, the validation works as intended.
```javascript
unique: {
	url: '/api/validators/unique'
},
```
But when setting it as an async function it always displays the error message
```javascript
unique: async (value, name, params, el$, form$) => {
	return 'true';
},
```
or
```javascript
  unique: async (value, name, params, el$, form$) => {
    const res = await axios.post('/api/validators/unique', {
      value,
      name,
      params,
    })

    return res.data // should be `true` or `false`
  }
```
Am I doing anything wrong? I'm currently using version 1.7.1

**Comment by adamberecz** [comment, marked as answer]
Yes, this was a bug and should be fixed in [f005f20](https://github.com/vueform/vueform/commit/f005f2001731cd8b6a75301eea3657a33a71fa54).

Until the patch is released the workaround is to return `{data: true}` instead of `true`.

**Reply by mikemamaril** [reply]
Thanks for this. It would be ideal if there would be a way to send different errors based on response like a 429 response code/message for throttled requests. I managed this by making custom validator instead of using the unique/exists rules.

**Reply by adamberecz** [reply]
It's an interesting proposition. Feel free to throw it in the [Ideas](https://github.com/vueform/vueform/discussions/categories/ideas).
